AaSystem/RequestManagment/RequestProcessor.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/python3
+from AaSystem.RequestManagment.CommandFileReferences import *
+from AaSystem.LogAndPrint.Log import PrintAndLog, PrintRedAndLog
+from AaSystem.EventQueue.CommandQueue import EmptyCommandQueue, DeQueueCommand, CommandQueueNotEmpty
+from AaSystem.RequestManagment.RequestBuilder import ParseRequest
+from AaSystem.Reflection.Reflect import GetPublicFacingFunctionsFromPath
+
+
+CommandMapping = {}
+CommandMappingNameByTree = {}
+
+
+def CreateCommandConnections():
+    for commandType in CommandTypes:
+        if commandType not in CommandMappingNameByTree:
+            CommandMappingNameByTree[commandType] = []
+        for commandPath in CommandTypes[commandType]:
+            commands = GetPublicFacingFunctionsFromPath(commandPath)
+            for command in commands:
+                CommandMapping[str.lower(command[1].PublicFacing)] = command[1]
+                CommandMappingNameByTree[commandType].append(command[1].PublicFacing)
+
+
+def RunCommand(request):
+    if not request:
+        PrintRedAndLog("Request cannot be null or empty")
+        return False
+
+    request = ParseRequest(request)
+
+    if len(request) < 1:
+        PrintRedAndLog("Request Missing parameters")
+        return False
+    requestedCommand = str.lower(request[0])
+    if requestedCommand == "-help":
+        BaseCommandsHelp()
+        return True
+    if requestedCommand not in CommandMapping:
+        PrintRedAndLog(f"No such supported command '{requestedCommand}'")
+        return False
+    return CommandMapping[requestedCommand](request[1:])
+
+
+def RunCommands():
+    while CommandQueueNotEmpty():
+        request = DeQueueCommand()
+        success = RunCommand(request)
+        if success is not None and not success:
+            EmptyCommandQueue()
+            return
+
+
+def BaseCommandsHelp():
+    PrintAndLog("\n"
```

Comment:
check argparse -  https://docs.python.org/3/library/argparse.html

Reply:
Yeah, I know about argparse, sometimes if feels a bit too constraining considering the information I want to present and how I want to present it when it comes to help functions and the cli supports dimensions of recursive instructions that I feel might conflict with how the argparse interpretes things